### PR TITLE
DS-3108 DSpace 6x: Support non-email based authentication in REST API

### PR DESCRIPTION
**Pull request by tomdesair**
This is the DSpace 6x version of https://github.com/DSpace/DSpace/pull/1598

### CODE REVIEW

File: dspace-rest/src/main/webapp/WEB-INF/security-applicationContext.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
         <property name="authenticationManager" ref="dspaceAuthenticationManager"/>
         <!-- Ensures that after login no redirect is made to the home page, the rest will return a 200 status code -->
         <property name="authenticationSuccessHandler" ref="org.dspace.rest.authentication.NoRedirectAuthenticationLoginSuccessHandler"/>
-        <property name="usernameParameter" value="email"/>
+        <property name="usernameParameter" value="user"/>
```
Comment (tdonohue) [inline dspace-rest/src/main/webapp/WEB-INF/security-applicationContext.xml]: To ensure we have no breaking changes between 6.0 and 6.1, I think we need to revert this back to `email` (and make appropriate changes to the other code).  I agree with the idea to switch this to `user` for 7.0, but as that's a breaking change, we should only do that in a major release.

**Review by tdonohue — DISMISSED** [review]

File: dspace-rest/src/main/webapp/WEB-INF/security-applicationContext.xml (hunk 1 above)
Comment (tomdesair) [inline dspace-rest/src/main/webapp/WEB-INF/security-applicationContext.xml]: You are right. I've reverted the rename of the username parameter in this pull request.

**Comment by tomdesair** [comment]
I've reverted the rename of the REST API login parameter email to user.

**Comment by terrywbrady** [comment]
+1, successfully tested on 6-x branch

**Review by bram-atmire — APPROVED** [review]
+1 tdonohue reviewed, left comments, resolved by Tom. Terry Brady has deployed & tested. Good to go.

**Pull request merged**